vendor/views/vendor_admin.py

```diff
@@ -104,8 +113,13 @@ def get_initial(self):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
-        context['formset'] = PriceFormSet(instance=self.object)
+        offer_products = self.object.products.all()
+        customer_who_own = set([customer_profile for customer_profile in CustomerProfile.objects.all() if customer_profile.has_product(offer_products) ])
```

Comment:
Can this be done with a query?  It seems that this could become very expensive with lots of CustomerProfiles.

Reply:
yes, changed to `CustomerProfile.objects.filter(receipts__products__in=offer_products)`